src/MerchantCenter/MerchantStatuses.php

```diff
@@ -354,67 +369,71 @@ protected function fetch_issues( string $type = null, int $per_page = 0, int $pa
+	 * Get MC product issues from a list of Product View statuses.
 	 *
-	 * @param string[] $google_ids
+	 * @param [] $statuses The list of Product View statuses.
 	 *
-	 * @return GoogleProductStatus[] Statuses found to be valid.
+	 * @return array The list of product issues.
 	 */
-	protected function filter_valid_statuses( array $google_ids ): array {
+	protected function get_product_issues( array $statuses ): array {
 		/** @var Merchant $merchant */
 		$merchant = $this->container->get( Merchant::class );
 		/** @var ProductHelper $product_helper */
 		$product_helper      = $this->container->get( ProductHelper::class );
 		$visibility_meta_key = $this->prefix_meta_key( ProductMetaHandler::KEY_VISIBILITY );
 
-		$valid_statuses = [];
+		$google_ids     = array_column( $statuses, 'mc_id' );
+		$product_issues = [];
+		$created_at     = $this->cache_created_time->format( 'Y-m-d H:i:s' );
 		foreach ( $merchant->get_productstatuses_batch( $google_ids )->getEntries() as $response_entry ) {
 			$mc_product_status = $response_entry->getProductStatus();
-			if ( ! $mc_product_status ) {
-				do_action(
-					'woocommerce_gla_debug_message',
-					'A Google ID was not found in Merchant Center.',
-					__METHOD__
-				);
-				continue;
-			}
-			$mc_product_id = $mc_product_status->getProductId();
-			$wc_product_id = $product_helper->get_wc_product_id( $mc_product_id );
-			// Skip products not synced by this extension.
-			if ( ! $wc_product_id ) {
-				continue;
-			}
-
-			// Product previously found/validated.
-			if ( ! empty( $this->product_data_lookup[ $wc_product_id ] ) ) {
-				$valid_statuses[] = $response_entry->getProductStatus();
-				continue;
-			}
+			$mc_product_id     = $mc_product_status->getProductId();
+			$wc_product_id     = $product_helper->get_wc_product_id( $mc_product_id );
+			$wc_product        = $this->product_data_lookup[ $wc_product_id ] ?? null;
 
-			$wc_product = $product_helper->get_wc_product_by_wp_post( $wc_product_id );
-			if ( ! $wc_product || 'product' !== substr( $wc_product->post_type, 0, 7 ) ) {
-				// Should never reach here since the products IDS are retrieved from postmeta.
+			// Skip products not synced by this extension.
+			if ( ! $wc_product ) {
 				do_action(
 					'woocommerce_gla_debug_message',
 					sprintf( 'Merchant Center product %s not found in this WooCommerce store.', $mc_product_id ),
 					__METHOD__ . ' in remove_invalid_statuses()',
 				);
 				continue;
 			}
+			// Unsynced issues shouldn't be shown.
+			if ( ChannelVisibility::DONT_SYNC_AND_SHOW === $wc_product->get_meta( $visibility_meta_key ) ) {
+				continue;
+			}
 
-			$this->product_data_lookup[ $wc_product_id ] = [
-				'name'       => get_the_title( $wc_product ),
-				'visibility' => get_post_meta( $wc_product_id, $visibility_meta_key ),
-				'parent_id'  => $wc_product->post_parent,
+			$product_issue_template = [
+				'product'              => html_entity_decode( $wc_product->get_name() ),
```

Comment:
Yep, it would be more accurate but I'd prefer to keep it since it's tied to the column name in `wp_gla_merchant_issues`.

See: https://github.com/woocommerce/google-listings-and-ads/blob/7f3e1f57887658cadcc38f6795a36d1fb7fb1ccb/src/DB/Table/MerchantIssueTable.php#L33